src/app/app.component.html

```diff
@@ -1,3 +1,9 @@
-<app-header></app-header>
-<router-outlet></router-outlet>
-<app-footer></app-footer>
+<ng-container *ngIf="shouldShowHeaderAndFooter()">
```

Comment:
don't use method in template. Find a way how to get rid of this issue, maybe some class property instead of method / observable value

Reply:
+ you can use ngIf + else structure instead of 2 ngIfs

Reply:
Fixed